lib/matplotlib/tests/test_contour.py

```diff
@@ -478,3 +478,49 @@ def test_contour_autolabel_beyond_powerlimits():
     ax.clabel(cs)
     # Currently, the exponent is missing, but that may be fixed in the future.
     assert {text.get_text() for text in ax.texts} == {"0.25", "1.00", "4.00"}
+
+
+def test_countourf_legend_elements():
+    from matplotlib.patches import Rectangle
+    x = np.arange(1, 10)
+    y = x.reshape(-1, 1)
+    h = x * y
+
+    cs = plt.contourf(h, levels=[10, 30, 50],
+                      colors=['#FFFF00', '#FF00FF', '#00FFFF'],
+                      extend='both')
+    cs.cmap.set_over('red')
+    cs.cmap.set_under('blue')
+    cs.changed()
```

Comment:
@greglucas I thought you put in something to auto-call `.changed`?

Reply:
I just took the base from the `extend` example here: https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.contour.html?highlight=contour#matplotlib.axes.Axes.contour

So it may very well be that it is not required (and I can update the example).

Reply:
It appears that `set_over` and `set_under` requires an explicit call to `changed`. (At least using inline plotting in Spyder.)

Reply:
Or rather, it is not strictly required for the test, but for the plot to look correct. So I may just as well remove it from the test as there is no image comparison.

Reply:
My bad, it is required for the tests as well. (Just noted that I still got found regions without it, but not the correct color information.)